src/arrayMethodJoin.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sep = typeof separator === 'undefined' ? ',' : separator;
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (element === null || typeof element === 'undefined') {
+        if (i === this.length - 1) {
+          break;
+        }
+        result += sep;
+      } else {
+        result += element;
+
+        if (i < this.length - 1) {
```

Comment:
Move this check and appending 'sep' to the end of the loop, outside of the else block.